dnf5-plugins/copr_plugin/copr.cpp

```diff
@@ -0,0 +1,286 @@
+/// TODO: copyright headers mention libdnf?
+
+
+#include "copr.hpp"
+
+#include <fnmatch.h>
+
+#include <cstdio>
+#include <iostream>
+#include <regex>
+
+
+namespace dnf5 {
+
+using namespace libdnf::cli;
+
+#define dnf5_command "dnf5"
+
+#define warning(fmt, ...)                  \
```

Comment:
I failed to find a built-in warning utility, is there some?

Reply:
@praiskup 
> I failed to find a built-in warning utility, is there some?

No. But I wanted to add. Thanks for the reminder.

In general, try to use type-safe code.
- C++ templates instead of macros
- `libdnf::utils::sformat` instead of `printf`

NOTE: `libdnf::utils::sformat` has a format string similar to Python formating. See https://en.cppreference.com/w/cpp/utility/format/format. Don't use `std::format`, in Fedora 37 is GCC12 and support for `std::format` starts from GCC13.

Example:
```
template <typename... Args>
void warning(const char * format, Args &&... args) {
    std::cerr << "WARNING: " + libdnf::utils::sformat(format, std::forward<Args>(args)...) << std::endl;
}
```